File: app/src/main/java/ru/magflayer/spectrum/presentation/common/android/navigation/router/MainRouter.java
package ru.magflayer.spectrum.presentation.common.android.navigation.router;

import ru.magflayer.spectrum.presentation.common.android.navigation.Screens;
import ru.terrakok.cicerone.Router;

public class MainRouter extends Router {

    public void openCameraScreen() {
        newRootScreen(Screens.CAMERA_SCREEN);
    }

    public void openHistoryScreen() {
        navigateTo(Screens.HISTORY_SCREEN);
    }

    public void openHistoryDetailsScreen(final String filePath) {
        navigateTo(Screens.HISTORY_DETAILS_SCREEN, filePath);
    }

}

File: app/src/main/java/ru/magflayer/spectrum/presentation/common/android/navigation/router/MainRouter.kt
package ru.magflayer.spectrum.presentation.common.android.navigation.router

import com.github.terrakok.cicerone.Router
import ru.magflayer.spectrum.presentation.common.android.navigation.MainScreens

class MainRouter : Router() {

    fun openCameraScreen() = newRootScreen(MainScreens.Camera)

    fun openHistoryScreen() = navigateTo(MainScreens.HistoryList)

    fun openHistoryDetailsScreen(filePath: String) =
        navigateTo(MainScreens.HistoryDetails(filePath))

}
